### PR TITLE
3.1 clean errors

### PR DESCRIPTION
**Pull request by emidoots**
This should be my last push for more clean/idiomatic errors from GLFW.

Please see the individual commit descriptions for more information. 

Alternative to #112 (fixed typo in commit message)

### CODE REVIEW

File: error.go [hunk 1]
```diff
@@ -75,7 +105,12 @@ func flushErrors() {
 func fetchError() error {
 	select {
 	case err := <-lastError:
-		return err
+		switch err.Code {
+		case outOfMemory, invalidEnum, invalidValue, notInitialized, noCurrentContext:
```
Comment (dmitshur) [inline error.go]: It seems a little odd these are not in the same order as above, but I guess that's fine.

**Comment by dmitshur** [comment]
LGTM.

**Comment by tapir** [comment]
:thumbsup: LGTM

**Comment by tapir** [comment]
should we add the error anlaysis here (http://www.glfw.org/docs/latest/group__errors.html) to go doc documentation?

File: error.go (hunk 1 above)
Comment (emidoots) [inline error.go]: Done.

**Comment by emidoots** [comment]
According to the doc link @tapir sent, `GLFW_PLATFORM_ERROR` is:

> A bug in GLFW or the underlying operating system. Report the bug to our issue tracker.

On that premise, I'll also turned that one into a panic.

**Comment by tapir** [comment]
The thing is now all the functions return an error and most of the errors are actually panics.
Looks a bit overdone. Ideas?

**Comment by emidoots** [comment]
> should we add the error anlaysis here (http://www.glfw.org/docs/latest/group__errors.html) to go doc documentation?

Done.

PTAL again everyone.

**Comment by emidoots** [comment]
> The thing is now all the functions return an error and most of the errors are actually panics.
> Looks a bit overdone. Ideas?

Let's not discuss this here: I'll create a new issue to discuss it.

**Comment by tapir** [comment]
LGTM

**Comment by dmitshur** [comment]
LGTM.

**Pull request merged**